crates/spirv-std/src/arch.rs

```diff
@@ -140,3 +142,17 @@ pub unsafe fn vector_insert_dynamic<T: Scalar, V: Vector<T, N>, const N: usize>(
 
     result
 }
+
+/// Fragment-shader discard.
+///
+/// Ceases all further processing in any invocation that executes it: Only
+/// instructions these invocations executed before [kill] have observable side
+/// effects.
```

Comment:
I think I would prefer to keep the comments about how OpKill corresponds to discard in GLSL, and OpDemoteToHelperInvocationEXT corresponds to discard in HLSL, as that's what a lot of people are familiar with and would help those people understand what the difference between the two is.